python/lsst/ts/ess/labjack/labjack_accelerometer_data_client.py

```diff
@@ -0,0 +1,462 @@
+    type: string
+  min_frequency:
+    description: >-
+        Approximate minimum frequency to report in the PSD (Hz).
+        A value larger than 0 may be used to increase the resolution
+        of the reported PSD.
+        Must be ≤ max_frequency / {MIN_FREQUENCY_RATIO=}.
+    type: number
+    default: 0
+    minimum: 0
+  max_frequency:
+    description: >-
+        Maximum frequency to report in the PSD (Hz).
+        If larger than the LabJack can handle then it is reduced.
+    type: number
+    minimum: 0
+  num_frequencies:
+    description: >-
+        Number of frequencies to report in the PSD.
+        Must be less than or equal to the length of the PSD array fields.
+    type: integer
+    minValue: 10
+    default: 200
+  analog_inputs:
+    description: >-
+        Analog inputs read for x, y, and z data.
+        0 for AIN0, 1 for AIN1, 2 or AIN2, etc.
+    type: array
+    minItems: 3
+    maxItems: 3
+    items:
+        type: integer
+        minimum: 0
+  scale:
+    description: Accelerometer scale (m/s2 per Volt).
+    type: number
+required:
+  - device_type
+  - connection_type
+  - identifier
+  - sensor_name
+  - location
+  - min_frequency
+  - max_frequency
+  - num_frequencies
+  - analog_inputs
+  - scale
+additionalProperties: false
+"""
+        )
+
+    def descr(self) -> str:
+        return f"identifier={self.config.identifier}"
+
+    async def run(self) -> None:
+        """Read and process data from the LabJack."""
+        await self.start_data_stream()
+        await asyncio.Future()
+
+    async def disconnect(self) -> None:
+        self.mock_stream_task.cancel()
+        if self.handle is not None:
+            await self.stop_data_stream()
+        await super().disconnect()
+
+    async def start_data_stream(self) -> None:
+        """Start the data stream from the LabJack."""
+
+        return await self.run_in_thread(
+            func=self._blocking_start_data_stream, timeout=COMMUNICATION_TIMEOUT
+        )
+
+    def start_mock_stream(self) -> None:
+        """Start a mock stream, since ljm demo mode does not stream."""
+        if self.simulation_mode == 0:
+            raise ValueError("start_mock_stream can only be called in simulation mode")
+
+        self.mock_stream_task.cancel()
+        self.mock_stream_task = asyncio.create_task(self._mock_stream())
+
+    async def _mock_stream(self) -> None:
+        """Pretend to stream data.
+
+        Stream self.mock_raw_1d_data.
+        If that is None then set it to a random array.
+        """
+        self.log.info("mock_stream begins")
+        if self.sampling_interval is None:
+            raise RuntimeError("Streaming not started")
+        try:
+            sleep_interval = self.sampling_interval * self.num_samples
+            if self.mock_raw_1d_data is None and self.make_random_mock_raw_1d_data:
+                # Generate random mock data
+                # using half the available scale of -10 to 10 volts
```

Comment:
Most doc strings and some inline comments end in a dot and some, like this one, doesn't. Please review all and make sure that they all end in a dot (or all don't but I'd prefer it if they do).

Reply:
OK. I took a pass. I want to continue to omit periods for comments that come after code, but those are few. I fixed the rest in code (omitting build files).